xml/src/main/java/org/ehcache/xml/ConfigurationTranslator.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.xml;
+
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.config.Configuration;
+import org.ehcache.config.ResourcePool;
+import org.ehcache.config.ResourcePools;
+import org.ehcache.core.internal.util.ClassLoading;
+import org.ehcache.xml.model.BaseCacheType;
+import org.ehcache.xml.model.CacheEntryType;
+import org.ehcache.xml.model.CacheType;
+import org.ehcache.xml.model.ConfigType;
+import org.ehcache.xml.model.ObjectFactory;
+import org.ehcache.xml.model.ResourcesType;
+import org.ehcache.xml.model.ServiceType;
+import org.w3c.dom.Element;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.transform.Source;
+import javax.xml.transform.stream.StreamSource;
+
+import static org.ehcache.core.spi.service.ServiceUtils.findAmongst;
+import static org.ehcache.xml.ConfigurationParser.CORE_SCHEMA_JAXB_MODEL_PACKAGE;
+import static org.ehcache.xml.ConfigurationParser.CORE_SCHEMA_URL;
+import static org.ehcache.xml.XmlConfiguration.CORE_CACHE_CONFIGURATION_PARSER;
+import static org.ehcache.xml.XmlConfiguration.CORE_SERVICE_CONFIGURATION_PARSERS;
+import static org.ehcache.xml.XmlConfiguration.CORE_SERVICE_CREATION_CONFIGURATION_PARSERS;
+
+public class ConfigurationTranslator {
+
+  private final Configuration configuration;
+  private final Marshaller marshaller;
+  private final Unmarshaller unmarshaller;
+
+  private final Map<Class<?>, CacheManagerServiceConfigurationParser> xmlParsers = new HashMap<>();
+  private final Map<Class<?>, CacheServiceConfigurationParser> cacheXmlParsers = new HashMap<>();
+  private final ResourceConfigurationParser resourceConfigurationParser;
+
+  public ConfigurationTranslator(Configuration configuration) throws JAXBException, IOException {
+    this.configuration = configuration;
+    JAXBContext jaxbContext = JAXBContext.newInstance(CORE_SCHEMA_JAXB_MODEL_PACKAGE);
+    marshaller = jaxbContext.createMarshaller();
```

Comment:
We should probably be validating the configuration we generate against the schema when marshalling.

Reply:
Will see how I can do that